### ISSUE TITLE
make error BLKGETZONESZ undeclared

### ISSUE THREAD

**Issue by Trina0224** [westerndigitalcorporation/libzbd#1]
Hi 
I compiled release version 1.1.0 and met this:
```
trina@b2-SYS-5019D-FN8TP:~/Downloads/libzbd-1.1.0$ make
Making all in .
make[1]: Entering directory '/home/trina/Downloads/libzbd-1.1.0'
  CC       lib/libzbd_la-zbd.lo
lib/zbd.c: In function ‘zbd_do_get_info’:
lib/zbd.c:225:18: error: ‘BLKGETZONESZ’ undeclared (first use in this function); did you mean ‘BLKGETSIZE64’?
  ret = ioctl(fd, BLKGETZONESZ, &zone_sectors);
                  ^~~~~~~~~~~~
                  BLKGETSIZE64
lib/zbd.c:225:18: note: each undeclared identifier is reported only once for each function it appears in
lib/zbd.c:239:18: error: ‘BLKGETNRZONES’ undeclared (first use in this function); did you mean ‘BLKGETZONESZ’?
  ret = ioctl(fd, BLKGETNRZONES, &nr_zones);
                  ^~~~~~~~~~~~~
                  BLKGETZONESZ
Makefile:779: recipe for target 'lib/libzbd_la-zbd.lo' failed
make[1]: *** [lib/libzbd_la-zbd.lo] Error 1
make[1]: Leaving directory '/home/trina/Downloads/libzbd-1.1.0'
Makefile:962: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1

``` 
My environment is Ubuntu 18.04, I can make libzbc with no problem, I think no libraries related issues.

Thanks

**Comment by floatious** [comment]
Hello Trina,

BLKGETZONESZ is defined in include/uapi/linux/blkzoned.h
It was added to Linux in commit 72cd87576d1d ("block: Introduce BLKGETZONESZ ioctl")
which was first included in kernel release v4.20 which was released Dec 23 2018.

Ubuntu 18.04 uses kernel v4.15, which is too old.

In Ubuntu, the Linux UAPI headers, like blkzoned.h is supplied by the package:
https://packages.ubuntu.com/search?searchon=contents&keywords=blkzoned.h&mode=exactfilename&suite=bionic&arch=any
linux-libc-dev

While Ubuntu does provide newer kernels for older LTS releases:
https://ubuntu.com/kernel/lifecycle

They only provide an update for the kernel itself.
For some reason, they do not provide an update for the Linux UAPI headers.

I suggest that you either:

1) Since you are running an LTS version, you could install Ubuntu 20.04 LTS,
which is the latest Ubuntu LTS.

2) Follow the steps in https://ubuntu.com/kernel/lifecycle
and then additionally install the Linx UAPI headers manually:
$ apt source linux-hwe-5.4-headers-5.4.0-42
$ cd linux-hwe-5.4-5.4.0
$ sudo make headers_install INSTALL_HDR_PATH=/usr
(If your distro later updates the linux-libc-dev package, it will
overwrite the Linux UAPI headers, so then you will need to
run the make headers_install step again.)

**Comment by Trina0224** [comment]
Thank you. It's working on new kernel as you mentioned.
